src/ARTED/common/hpsi.f90

```diff
@@ -150,34 +155,54 @@ subroutine subtraction(Vloc,tpsi,htpsi,ttpsi)
       end do
     end subroutine
 
-    subroutine pseudo_pt(ik,tpsi,htpsi)
-#ifdef ARTED_STENCIL_PADDING
-      use opt_variables, only: zJxyz => zKxyz
-#else
-      use opt_variables, only: zJxyz
-#endif
+    !Calculating nonlocal part
+    subroutine pseudo_pt(ik,idx_proj,zproj,tpsi,htpsi)
+      use opt_variables, only: NP,NPI,pseudo_start_idx
+      use global_variables, only: NI,Nps
       implicit none
       integer,    intent(in)  :: ik
+      integer,    intent(in)  :: idx_proj(NPI)
+      complex(8), intent(in)  :: zproj(Nps,Nlma)
       complex(8), intent(in)  :: tpsi(0:PNL-1)
       complex(8), intent(out) :: htpsi(0:PNL-1)
-      integer    :: ilma,ia,j,i
+
+      integer    :: ia,i,j,ip,ioffset
       complex(8) :: uVpsi
 
-      !Calculating nonlocal part
-      do ilma=1,Nlma
-        ia=a_tbl(ilma)
-        uVpsi=0.d0
+      complex(8) :: pseudo(NPI)
+      complex(8) :: dpseudo(NPI)
+
+      dpseudo = cmplx(0.d0)
+
+      ! gather (load) pseudo potential point
+      do i=1,NPI
+        pseudo(i) = tpsi(idx_proj(i))
```

Comment:
I think that some parts of `idx_proj(i)` are indefinite since the first dimension of `zJxyz(:,:)` is allocated up to `Nps` but initialized only up to `Mps(ia)` for each ion. Thus, if `Mps(ia)` has different value for different ions, the use of the full array `idx_proj(1:NPI=Nps*NI)` may induce segmentation fault.

Reply:
Thank you for reviewing.
I applied your suggestion by the commit ade3d81 that this commit removes uninitialized elements.